examples/tasks/async_compute.rs

```diff
@@ -0,0 +1,114 @@
+use bevy::{prelude::*, tasks::{AsyncComputeTaskPool, Task}};
+use std::time::{Instant, Duration};
+use rand::Rng;
+use futures_lite::future;
+
+/// This example shows how to use the ECS and the AsyncComputeTaskPool
+/// to spawn, poll, and complete tasks across systems and system ticks.
+
+// Number of cubes to spawn across the x, y, and z axis
+const NUM_CUBES: u32 = 6;
+
+/// This system generates tasks simulating computationally intensive
+/// work that potentially spans multiple frames/ticks. A separate
+/// system, handle_tasks, will poll the spawned tasks on subsequent
+/// frames/ticks, and use the results to spawn cubes
+fn spawn_tasks(
+    mut commands: Commands,
+    thread_pool: Res<AsyncComputeTaskPool>,
+) {
+    for x in 0 ..NUM_CUBES {
+        for y in 0 ..NUM_CUBES {
+            for z in 0 ..NUM_CUBES {
+
+                // Spawn new task on the AsyncComputeTaskPool
+                let task = thread_pool.spawn(async move {
+
+                    let mut rng = rand::thread_rng();
+                    let start_time = Instant::now();
+                    let duration = Duration::from_secs_f32(rng.gen_range(0.05..0.2));
+                    while Instant::now() - start_time < duration {
+                        // Spinning for 'duration', simulating doing hard
+                        // compute work generating translation coords!
+                    }
+
+                    // Such hard work, all done!
+                    println!("Done generating translation coords x: {} y: {} z: {}", x, y, z);
+
+                    Transform::from_translation(Vec3::new(x as f32, y as f32, z as f32))
+                });
+
+                // Spawn new entity and add our new task as a component
+                commands.spawn().insert(task);
+            }
+        }
+    }
+}
+
+/// This system queries for entities that have our Task<Transform> component. It polls the
+/// tasks to see if they're complete. If the task is complete it takes the result, adds a
+/// new PbrBundle of components to the entity using the result from the task's work, and
+/// removes the task component from the entity.
+fn handle_tasks(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut our_entity_tasks: Query<(Entity, &mut Task<Transform>)>
+) {
+    our_entity_tasks.for_each_mut(|(entity, mut task)| {
+
+        if let Some(transform) = future::block_on(future::poll_once(&mut *task)) {
+
+            // Normally we would add our mesh and material assets once
+            // and store the handle, but that's for another example
+            let box_mesh = meshes.add(Mesh::from(shape::Cube { size: 0.25 }));
+            let box_material = materials.add(Color::rgb(1.0, 0.2, 0.3).into());

```

Comment:
Given how common copy-pasting example code is, I think we should strive to remain "correct" in all examples.

Reply:
`box_mesh` and `box_material` are the same for every cube, they should be created outside of the loop and the handles reused

Reply:
> `box_mesh` and `box_material` are the same for every cube, they should be created outside of the loop and the handles reused... actually as you said in the comment

Is there an idiomatic "bevy" way of storing the handle? In my own application I followed how Bevy does it with its default Pbr Pipeline Handle, creating a const handle:

https://github.com/bevyengine/bevy/blob/e3435e5144b79908d22eabcefb7bee92d5f520ba/crates/bevy_pbr/src/render_graph/pbr_pipeline/mod.rs#L13-L14

Reply:
Storing it as a resource is the most common approach.

Reply:
> Storing it as a resource is the most common approach.
 
Wrap it in a tuple struct? Or regular struct with a field?

Reply:
Changed in bd73bee